src/Form/tests/FormControl.test.jsx

```diff
@@ -31,4 +49,21 @@ describe('FormControl', () => {
     expect(onChangeFunc).toHaveBeenCalledTimes(inputText.length);
     expect(ref.current.style.height).toEqual(`${ref.current.scrollHeight + ref.current.offsetHeight}px`);
   });
+
+  it('should apply and accept input mask for phone numbers', () => {
+    render(<Component />);
+
+    act(() => {
+      const input = screen.getByTestId('form-control-with-mask');
+      userEvent.type(input, '1234567890');
+      expect(input.value).toBe('+1(234)567-89-0');
+    });
+  });
+  it('should be cleared from the mask elements value', () => {
+    render(<Component isClearValue />);
+
+    const input = screen.getByTestId('form-control-with-mask');
+    fireEvent.change(input, { target: { value: '1234567890' } });
```

Comment:
[curious] Why use `userEvent.type` but `fireEvent.change` here? Ideally, we could rely on `userEvent` to more closely mimic real user behavior.